src/components/GraphUtils.js

```diff
@@ -0,0 +1,202 @@
+import * as d3 from 'd3';
+
+export function truncate(fullStr, strLen, separator='...') {
```

Comment:
This function is not graph-related. Consider extracting it into a separate module (e.g., `Utils.js`).